feat(service): add sftpgo template

vincentbaeten opened the pull request:
## Changes

- feat(service): add [SFTPGo](https://sftpgo.com/) service template

vincentbaeten committed:
Add SFTPGo service template

Cinzya commented on templates/compose/sftpgo.yaml:
```diff
@@ -0,0 +1,44 @@
+      postgresql:
+        condition: service_healthy
+    ports:
+      - 2022:2022
```
Is this port mapping necessary? We want to avoid mapping any ports if possible for security reasons.

vincentbaeten replied:
It is needed for the SFTP server part but also not strictly needed because users could also use it with the web client only to upload and manage files. AFAIK there isn't some conditional ability to toggle the port mapping on/off? Would it be best to have separate service templates with/without port mapping or put an extra note on the service documentation page?

Cinzya replied:
We could put it into an env variable and mark it as required without a default value. This will prompt the user to fill it out before starting the service and then they can decide themselves what port to map.
It's not in the docs yet, but this issue (https://github.com/coollabsio/coolify-docs/issues/345) roughly explains how to make a required env variable.

Cinzya commented on templates/compose/sftpgo.yaml:
```diff
@@ -0,0 +1,44 @@
+# logo: svgs/sftpgo.svg
+# port: 8080
+
+version: 3.8
```
AFAIK you don't need this line in Coolify. the others can correct me if I'm wrong.

vincentbaeten replied:
I will remove it and commit the change. It seems that it is indeed no longer necessary in docker compose files starting from v2. Didn't know this.

vincentbaeten committed:
Update SFTPGo service template

- Add SFTP port mapping as required env variable
- Add logging level as env variable with default as "info"
- Remove "version" key
- Fix logo extension

vincentbaeten committed:
Add volume for SFTPGo generated keys

0xtlt commented:
really need this

peaklabs-dev committed:
Merge branch 'next' into v4.x

github-actions[bot] committed:
docs: update changelog

peaklabs-dev committed:
feat(service): improve sftpgo

- update sftpgo to v2.7
- change port mapping syntax as it caused issues
- fix formatting, ordering and naming

github-actions[bot] committed:
docs: update changelog

peaklabs-dev commented:
Thanks for the PR. I upgraded SFTPGo to v2.7 and it will be available in the next version.

Boardiesman commented on templates/compose/sftpgo.yaml:
```diff
@@ -1,42 +1,42 @@
-      - SFTPGO_DATA_PROVIDER__CONNECTION_STRING=postgresql://${SERVICE_USER_POSTGRES}:${SERVICE_PASSWORD_64_POSTGRES}@postgresql:5432/${POSTGRES_DATABASE:-sftpgo}
-      - SFTPGO_SFTPD__BINDINGS__0__PORT=${SFTP_0_PORT?Port is required for the first SFTP binding}
+      - SFTPGO_DATA_PROVIDER__CONNECTION_STRING=postgresql://${SERVICE_USER_POSTGRES}:${SERVICE_PASSWORD_64_POSTGRES}@postgresql:5432/${POSTGRES_DATABASE:-sftpgo-db}
+      - SFTPGO_SFTPD__BINDINGS__0__PORT=2022
```
Port should be 2222 to match docker ports. Service is otherwise inaccessible on port 2222.

peaklabs-dev replied:
Thanks for the point fixed it on `next`.
